engine/platform/win32/con_win.c

```diff
@@ -573,10 +590,21 @@ void Wcon_DestroyConsole( void )
 
 	Sys_CloseLog();
 
-	if( s_wcd.hWnd )
+	if( !s_wcd.attached )
+	{ 
+		if( s_wcd.hWnd )
+		{
+			ShowWindow( s_wcd.hWnd, SW_HIDE );
+			s_wcd.hWnd = 0;
+		}
+	}
+	else
 	{
-		ShowWindow( s_wcd.hWnd, SW_HIDE );
-		s_wcd.hWnd = 0;
+		// reverts title & code page for console window that was before starting Xash3D
+		SetConsoleCP( s_wcd.previousCodePage );
+		SetConsoleOutputCP( s_wcd.previousOutputCodePage );
+		SetConsoleTitle( &s_wcd.previousTitle );
+		Con_Reportf( "Press Enter to continue...\n" );
```

Comment:
Maybe this should be Con_Printf, in case if our developer level is 0?

Other than that, I'm fine with these changes.

Reply:
Ok, I changed it to `Con_Printf`